crates/ere-hosts/src/main.rs

```diff
@@ -0,0 +1,76 @@
+//! Binary for benchmarking different Ere compatible zkVMs
+
+use anyhow::Result;
+use rayon::prelude::*;
+use std::sync::Arc;
+use std::{collections::HashMap, path::PathBuf};
+use witness_generator::generate_stateless_witness;
+use zkevm_metrics::WorkloadMetrics;
+use zkvm_interface::{Compiler, Input, zkVM};
+
+/// Main entry point for the host benchmarker
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    dotenv::dotenv().ok();
+
+    // SP1 zkVM
+    benchmark::<ere_sp1::RV32_IM_SUCCINCT_ZKVM_ELF, ere_sp1::EreSP1>(
+        "sp1",
+        concat!(env!("CARGO_WORKSPACE_DIR"), "ere-guests/sp1"),
+    )?;
+
+    // TODO: Add more backends
+    Ok(())
+}
+
+// TODO: Eventually move this into benchmark_runner
+fn benchmark<C, V>(host_name: &str, guest_dir: &str) -> Result<()>
+where
+    C: Compiler + Send + Sync,
+    C::Error: std::error::Error + Send + Sync + 'static,
+    V: zkVM<C> + Sync,
+    V::Error: std::error::Error + Send + Sync + 'static,
+{
+    println!("Benchmarking `{}`…", host_name);
+
+    // Compile program and create proving/verification keys
+    let program = C::compile(&PathBuf::from(guest_dir))?;
+    let zkvm_instance = V::new(program);
+    let zkvm_ref = Arc::new(&zkvm_instance);
+
+    let corpuses = generate_stateless_witness::generate();
+
+    // Iterate through test corpus and generate reports
+    // TODO: note that when proving, processing these in parallel will likely skew the results
```

Comment:
We can probably switch to serial processing and get a powerful server to run these on